examples/simple_test.go

```diff
@@ -0,0 +1,42 @@
+package examples
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/tetratelabs/wazero/wasm"
+	"github.com/tetratelabs/wazero/wasm/wazeroir"
+)
+
+// Test_Simple implements a basic function in go ("hello"), called by a function defined Web Assembly ("run"):
+func Test_Simple(t *testing.T) {
+	// Decode simple.wasm which was pre-compiled from this text format:
+	//	(module
+	//		(import "" "hello" (func $hello))
+	//		(func (export "run") (call $hello))
+	//	)
+	buf, err := os.ReadFile("wasm/simple.wasm")
+	require.NoError(t, err)
+	mod, err := wasm.DecodeModule(buf)
+	require.NoError(t, err)
+
+	// Create a new store and add the function "hello" which the module imports
+	store := wasm.NewStore(wazeroir.NewEngine())
+	hello := func(_ *wasm.HostFunctionCallContext) {
+		fmt.Println("Hello from Wazero!")
+	}
```

Comment:
```suggestion
	var called bool
	hello := func(_ *wasm.HostFunctionCallContext) {
		called = false
	}
	
	// ...
	
	require.True(t, called)
```

Reply:
I meant I want to ensure the host function is actually called! :D

Reply:
ah good point! right now it is only a visual cue.